File: crmessenger-whatsapp/internal/model/sql-model.go
package model

var CreateTable = `
CREATE TABLE IF NOT EXISTS userbots (
    wa_phone_number TEXT PRIMARY KEY,
    
    account_id TEXT NOT NULL,
    
    created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
    updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP
);

CREATE TABLE IF NOT EXISTS amocrm_sources (
    amocrm_source_id INTEGER PRIMARY KEY,

    userbot_wa_phone_number TEXT NOT NULL,
    
    amocrm_pipeline_id INTEGER NOT NULL,
    amocrm_external_id TEXT NOT NULL,
    amocrm_scope_id TEXT NOT NULL,
    
    created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
    updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP
);

CREATE TABLE IF NOT EXISTS amocrm_contacts(
    id SERIAL PRIMARY KEY,
    
    wa_phone_number TEXT NOT NULL,
    amocrm_contact_id INTEGER NOT NULL,
    
    created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
    updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP
);
       
CREATE TABLE IF NOT EXISTS amocrm_chats (
    id SERIAL PRIMARY KEY,
    
    contact_id INTEGER NOT NULL,
    amocrm_conversation_id TEXT NOT NULL,
    amocrm_chat_id TEXT NOT NULL,
    
    created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
    updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP
);

CREATE TABLE IF NOT EXISTS amocrm_messages (
    id SERIAL PRIMARY KEY,
    chat_id INTEGER NOT NULL,
    
    amocrm_message_id TEXT NOT NULL,
    role TEXT NOT NULL,
    text TEXT NOT NULL,
    
    created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
    updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP
);

CREATE OR REPLACE FUNCTION update_updated_at()
RETURNS TRIGGER AS $$
BEGIN
    NEW.updated_at = NOW();
    RETURN NEW;
END;
$$ LANGUAGE 'plpgsql';
   
CREATE TRIGGER update_updated_at_trigger
BEFORE UPDATE ON userbots
FOR EACH ROW
EXECUTE PROCEDURE update_updated_at();

CREATE TRIGGER update_updated_at_trigger
BEFORE UPDATE ON amocrm_sources
FOR EACH ROW
EXECUTE PROCEDURE update_updated_at();

CREATE TRIGGER update_updated_at_trigger
BEFORE UPDATE ON amocrm_chats
FOR EACH ROW
EXECUTE PROCEDURE update_updated_at();

CREATE TRIGGER update_updated_at_trigger
BEFORE UPDATE ON amocrm_messages
FOR EACH ROW
EXECUTE PROCEDURE update_updated_at();

CREATE TRIGGER update_updated_at_trigger
BEFORE UPDATE ON amocrm_contacts
FOR EACH ROW
EXECUTE PROCEDURE update_updated_at();
`

var DropTable = `
DROP TABLE IF EXISTS userbots;
DROP TABLE IF EXISTS amocrm_sources;
DROP TABLE IF EXISTS amocrm_contacts;
DROP TABLE IF EXISTS amocrm_chats;
DROP TABLE IF EXISTS amocrm_messages;
`

File: crmessenger-authorization/internal/model/sql_model.py
create_account_table = """
CREATE TABLE IF NOT EXISTS accounts (
    id SERIAL PRIMARY KEY,
    
    account_id INTEGER NOT NULL UNIQUE,
    two_fa_status BOOLEAN DEFAULT FALSE,
    two_fa_key TEXT DEFAULT '',
    refresh_token TEXT DEFAULT '',
    
    created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
    updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP
);
"""

on_update_table_query1 = """
CREATE OR REPLACE FUNCTION update_updated_at()
RETURNS TRIGGER AS $$
BEGIN
    NEW.updated_at = NOW();
    RETURN NEW;
END;
$$ LANGUAGE 'plpgsql';
"""

on_update_table_query2 = """
CREATE TRIGGER update_updated_at_trigger
BEFORE UPDATE ON accounts
FOR EACH ROW
EXECUTE PROCEDURE update_updated_at();
"""
drop_account_table = """
DROP TABLE IF EXISTS accounts;
"""

create_queries = [
    create_account_table,
    on_update_table_query1,
    on_update_table_query2
]
drop_queries = [drop_account_table]
